File: tests/benchmarks/publish/python/fib.py
# Naive recursive Fibonacci. Same algorithm and size as fib.flow.
import time


def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)


def main():
    t0 = time.perf_counter()
    result = fib(35)
    secs = time.perf_counter() - t0
    print("result %d" % result)
    print("seconds %.9f" % secs)


if __name__ == "__main__":
    main()

File: tests/benchmarks/publish/c/fib.c
/* Naive recursive Fibonacci. Same algorithm and size as fib.flow. */
#include <stdio.h>
#include <stdint.h>
#include <time.h>

static int64_t fib(int n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(void) {
    uint64_t t0 = clock_gettime_nsec_np(CLOCK_MONOTONIC);
    int64_t result = fib(35);
    uint64_t t1 = clock_gettime_nsec_np(CLOCK_MONOTONIC);
    double secs = (t1 - t0) / 1e9;
    printf("result %lld\n", (long long)result);
    printf("seconds %.9f\n", secs);
    return 0;
}
